page-tracking/src/page_info.rs

```diff
@@ -164,7 +171,7 @@ impl PageInfo {
                 }
             }
             ConvertedLocked => Err(PageTrackingError::PageLocked),
-            Shared(rc) => {
+            Shared(rc) | BlockedShared(rc, _) => {
```

Comment:
so there's a bit of a wrinkle here i just realized...

we have two ways to end up with a `SharedBlock` page in `release()`:
1. a TVM that had the page blocked was killed before the page was unblocked or removed
2. a TVM that had the shared page mapped (while it was blocked in another TVM) was killed

the behavior here is correct for case (1). for case (2), we want to leave it as `BlockedShared` but decrement the reference count. seems like we need to know whether or not the page was invalidated (indicating it was blocked) when releasing the page in order to differentiate between the two cases.

Reply:
I see... Should we carry the TVM id through the `BlockedShared()` state? That could help us know what to do. Or did you have another idea about how to do it?

Reply:
i did think about that, but it grows the `PageInfo` struct by 64-bits. that's a non-trivial amount of overhead considering the struct is repeated for every page of memory in the system.

another option is to have separate, e.g.`release_present()` vs `release_invalidated()`, calls for the two paths.

Reply:
>another option is to have separate, e.g.release_present() vs release_invalidated(), calls for the two paths.

Ok I'm gonna try this approach

Reply:
I've added the commit https://github.com/rivosinc/salus/pull/261/commits/f1bdd023ea5d41f82b377909d1724c664ad54e66 to handle this issue, please let me know if this is covering what you had in mind.

Reply:
yes, that looks right to me. squash with this commit?